src/libproxy/proxy.c

```diff
@@ -48,6 +48,7 @@ px_proxy_factory_new (void)
 pxProxyFactory *
 px_proxy_factory_copy (pxProxyFactory *self)
 {
+  g_object_ref (self->manager);
```

Comment:
Please swap those lines and make it clear with a new value, e.g. `new_proxy_factory` that is a new object.